jetty-core/jetty-http/src/test/java/org/eclipse/jetty/http/HttpURITest.java

```diff
@@ -641,6 +642,91 @@ public void testPathQuery(String input, String canonicalPath, String decodedPath
         assertThat(uri.hasAmbiguousEncoding(), is(expected.contains(Violation.AMBIGUOUS_PATH_ENCODING)));
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {
+        "/a%2Fb",
+        "/a%2F",
+        "/%2f",
+        "/%2f/"
+    })
+    public void testAmbiguousViaBuilderPath(String input)
+    {
+        HttpURI uri = HttpURI.build().path(input);
+        assertThat("has any violation", uri.hasViolations(), is(true));
+        assertThat("is ambiguous", uri.isAmbiguous(), is(true));
+    }
+
+    public static Stream<Arguments> suspiciousPathCharacterData()
+    {
+        return Stream.of(
+            // backslash
+            Arguments.of("/a%5Cb"),
+            Arguments.of("/foo/bar/zed/%5c"),
+            Arguments.of("/foo/bar/zed/%5C"),
```

Comment:
The spec says: 
> The "\" character encoded or not. (e.g. /path\info)
So can we add some tests for non encoded \ ?

Reply:
Done.   pushed a fix to make the semantic more in line with what the Servlet spec says.